pallets/simple-staking/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+
+
+	#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug)]
+	pub struct AuthorInfo<AccountId, Balance, BlockNumber> {
+		pub who: AccountId,
+		pub deposit: Balance,
+		pub last_block: BlockNumber
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	// The pallet's runtime storage items.
+	// https://substrate.dev/docs/en/knowledgebase/runtime/storage
+	
+	#[pallet::storage]
+	#[pallet::getter(fn invulnerables)]
+	pub type Invulnerables<T: Config>= StorageValue<_, Vec<T::AccountId>>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn authors)]
+	pub type Authors<T: Config> = StorageValue<_, Vec<AuthorInfo<T::AccountId, BalanceOf<T>, T::BlockNumber>>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn max_authors)]
+	pub type MaxAuthors<T> = StorageValue<_, u32>; 
+
+	#[pallet::storage]
+	#[pallet::getter(fn authority_bond)]
+	pub type AuthorityBond<T: Config>= StorageValue<_, BalanceOf<T>>;
+
+	
+	#[pallet::event]
+	#[pallet::metadata(T::AccountId = "AccountId")]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// Event documentation should end with an array that provides descriptive names for event
+		/// parameters. [something, who]
+		NewInvulnerables(Vec<T::AccountId>),
+		NewMaxAuthorCount(u32),
+		AuthorAdded(T::AccountId, BalanceOf<T>),
+		AuthorRemoved(T::AccountId)
+	}
+
+	// Errors inform users that something went wrong.
+	#[pallet::error]
+	pub enum Error<T> {
+		MaxAuthors,
+		Unknown,
+		Permission,
+		AlreadyAuthor,
+		NotAuthor
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		//TODO on init (or finalize) add to aura set at next era 
+		//TODO split half the pot to the author per block
+	}
+
+	// Dispatchable functions allows users to interact with the pallet and invoke state changes.
+	// These functions materialize as "extrinsics", which are often compared to transactions.
+	// Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+
+
+		#[pallet::weight(10_000)]
+		pub fn set_invulnerables(origin: OriginFor<T>, new: Vec<T::AccountId>) -> DispatchResultWithPostInfo {
+			//TODO chose protective scheme requires message from relay chain 
+			ensure_root(origin)?;
+			<Invulnerables<T>>::put(&new);
+			Self::deposit_event(Event::NewInvulnerables(new));
+			Ok(().into())
+		}
+
+		#[pallet::weight(10_000)]
+		pub fn set_max_author_count(origin: OriginFor<T>, max_authors: u32) -> DispatchResultWithPostInfo {
+			//TODO chose protective scheme requires message from relay chain 
+			ensure_root(origin)?;
+			<MaxAuthors<T>>::put(&max_authors);
+			Self::deposit_event(Event::NewMaxAuthorCount(max_authors));
+			Ok(().into())
+
+		}
+
+		#[pallet::weight(10_000)]
+		pub fn author_intent(origin: OriginFor<T>, deposit: BalanceOf<T>) -> DispatchResultWithPostInfo {
+			// lock deposit to start or require min?
```

Comment:
I would change a few things in this function, but defer to @kianenigma for his advice.

1. In the args, do we even need a `deposit` param? I think we can add a storage item for `AuthorBond` and just take it.
2. Let's use a lock and not reserve.
3. Change the name, since one is not "authoring intent"; I'd go with `register_as_author()`.
4. Make `last_block` an `Option` and initialize it to `None`. We shouldn't give someone a freebie.
5. I don't think we should return an error if the author exists. It could be that they exist with a bond of 100, and we raised it to 1000, and they just call this function again to up their deposit. The `true` case should check the actual locked vs. needed and (un)lock the difference.
6. Finally, emit an event of `AuthorRegistered(AccountId, Balance /*new, not total, amount locked*/)`

Reply:
yup this makes sense, as I was writting it I was going under the assumption we wanted to have the bidding for slots approach, but then wasn't quite sure. I also went with the assumption that if someone gets a cheap slot it is theirs even if it moves up, all can be changed, just was the assumption I was going under to start

Reply:
Agree with 1, 2, 3, 4, 6

2: I would consider reserving as well, as it is stricter and allows funds to not be used for anything else. I doubt if statemint would have governance or other functionality in which collators could use their tokens? 

5: I would actually keep it simple, return error. If someone wants to update, they should leave and join again. But both work, I haven't read the code yet.

Also not that locks are not slashable, reserved balance is.

Reply:
after thinking about the new design, I agree with Joe here I think. If someone has to renounce themselves then rejoin as the deposit increases they may lose their slot. Instead I can be if they are an author and their deposit is less then the current deposit, then reserve the difference, then if not error

Reply:
I don't have a strong preference on lock vs. reserve. Proxy and Multisig both reserve, so those are exclusive anyway. Registering an asset also reserves. So it's not like we have governance or inflationary rewards to lock for. If reserve keeps things simpler, I'm OK with it.